mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,376 @@
 ===========================================
 
-.. include:: /../common/stub-notice.txt
+By `Rick Anderson`_
+
+    - `Getting started with Tag Helpers`_
+    - `Starting the email tag helper`_
+    - `A working email tag helper`_
+    - `The bold tag helper`_
+    - `Web site information tag helper`_
+    - `Condition tag helper`_
+    - `Avoiding tag helper conflicts`_
+    - `Inspecting and retrieving child content`_
+    - `Wrap up and next steps`_
+
+You can browse the source code for the sample app used in this document on `GitHub <https://github.com/aspnet/Docs/tree/master/mvc/views/tag-helpers/authoring/sample>`__. 
+
+Getting started with tag helpers
+------------------------------------
+
+This tutorial provides an introduction to programming tag helpers. :doc:`intro` describes the benefits that tag helpers provide.
+
+A tag helper is any class that implements the `ITagHelper <https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ITagHelper.cs>`__ interface. However, when you author a tag helper, you generally derive from `TagHelper <https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers>`__ because doing so gives you access to the methods and properties that are required. We will introduce the ``TagHelper`` methods and properties as we use them in this tutorial.
+
+#. Create a new ASP.NET MVC 6 project called **AuthoringTagHelpers**. You won't need authentication for this project.
+#. Create a folder to hold the tag helpers called *TagHelpers*. The *TagHelpers* folder is not required, but it is a reasonable convention. Now let's get started writing some simple tag helpers.
+
+Starting the email tag helper
+--------------------------------
+
+In this section we will write a tag helper that updates an email tag. For example:
+
+``<email>Support</email>``
+
+The server will use our email tag helper to convert that markup into the following:
+
+.. code-block:: HTML
+
+	<a href="mailto:Support@example.com">Support@example.com</a>
+
+That is, an anchor tag that makes this an email link. You might want to do this if you are writing a blog engine and need it to send email for marketing, support, and other contacts, all to the same domain.
+
+#. Add the following ``EmailTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/z1EmailTagHelperCopy.cs
+   :language: c#
+   :lines: 2-16
+   
+Notes: 
+
+- The ``TargetElement`` attribute identifies "email" as the element to target for processing. If you don't specify a ``TargetElement`` attribute, ASP.NET MVC 6 uses a naming convention that targets elements of the root class name (minus the *TagHelper* portion of the class name). You could comment out this attribute (because the class is named **Email**\TagHelper), and it would target email tags. Using the attribute does make the intention explicit (that is, we are targeting **<email>** tags).
```

Comment:
The element name convention is intended to cover the most common case. We should lead with the convention and then tell people how to override it. I.e. the convention should be covered first in a first-class way and not be covered in a side note. Currently the note text makes it sound like using the convention is not recommended, which isn't the case.